apps/examples/smartfs/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,461 @@
+
+			for (i = 1; i <= ITR_READ; i++) {
+				fd = open(file[0], O_RDOK);
+				if (fd < 0) {
+					printf("Unable to open file: %s to read, fd: %d\n", file[0], fd);
+					close(frt_fd);
+					return ERROR;
+				}
+
+				do {
+				
+					ret = read(fd, read_buf, b_sz);
+				} while (ret == b_sz);
+
+				close(fd);
+			}
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+			ioctl(frt_fd, TCIOC_STOP, 0);
+			close(frt_fd);
+
+			tot_time += (after.timeleft - before.timeleft);
+			printf("End File size: %d bytes, Buffer size: %d bytes, Total time: %lu us\n", file_size, b_sz, tot_time);
+
+			ret = unlink(file[0]);
+			if (ret < 0) {
+				printf("Failed to unlink file %s, errno: %d\n", file[0], errno);
+				return ERROR;
+			}
+		}
+		break;
+
+	case TEST_FLAG_GC_EFFECT :
+		for (i = 1; i <= ITR_GC; i++) {
+			frt_fd = open(timer_path, O_RDONLY);
+			if (frt_fd < 0) {
+				printf("Unable to open file: %s, errno: %d\n", timer_path, errno);
+				return ERROR;
+			}
+
+			ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN);
+			ioctl(frt_fd, TCIOC_START, TRUE);
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before);
+
+			fd = open(file[0], O_WROK | O_CREAT);
+			if (fd < 0) {
+				printf("Unable to open file: %s, fd: %d\n", file[0], fd);
+				close(frt_fd);
+				return ERROR;
+			}
+
+			ret = write(fd, rw_buf, f_sz);
+			if (ret != (b_sz)) {
+				 printf("Unable to write to file: %s, ret: %d\n", file[0], ret);
+				 close(frt_fd);
+				 close(fd);
+				 return ERROR;
+			}
+			close(fd);
+
+			ret = unlink(file[0]);
+			if (ret != OK) {
+				printf("Failed to unlink file: %s, errno: %d\n", file[0], errno);
+				close(frt_fd);
+				return ERROR;
+			}
+
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+			ioctl(frt_fd, TCIOC_STOP, 0);
+			close(frt_fd);
+
+			itr_time = after.timeleft - before.timeleft;
+			printf("iteration time: %lu us\n", i, itr_time);
+			tot_time += itr_time;
+		}
+		printf("\nFinal time: %lu us\n", tot_time);
+		break;
+
+	default:
+		break;
+	}
+
+	return ret;
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int fs_performance_main(int argc, char *argv[])
+#endif
+{
+
+	int option = -1;
+	int ret = 0;
+	int timer_devno = 0;
+	f_sz = MAX_FILE_SIZE;
+	b_sz = MAX_RD_BUF_SIZE;
+
+	if (argc > 4) {
+		timer_devno = atoi(argv[1]);
+		option = atoi(argv[2]);
+		f_sz = atoi(argv[3]);
+		if (argc == 5) {
+			b_sz = atoi(argv[4]);
+		}
+	} else {
+		printf("Command Usage: fs_performance <Timer Device no.> <Option no.> <File Size in bytes> <Buffer Size in bytes>\n");
+		return -1;
+	}
+
+	snprintf(timer_path, _POSIX_PATH_MAX, TIMER_DEVNAME, timer_devno);
+	snprintf(file[0], TEST_FILE_NAME_LEN_MAX, "%s%d", TEST_FILE, 0);
+
+	switch (option) {
+		
+		case 1: goto FILE_CREATION_TEST;
+
+		case 2: goto OVERWRITE_TEST;
+
+		case 3: goto APPEND_TEST;
+
+		case 4: goto READ_TEST;
+
+		case 5: goto GC_TEST;
+
+		default: break;
```

Comment:
Can you add "goto errout" as the default case so the application exits.

Also, before exiting, you can make the application echo the correcf generic usage command for the app

Reply:
Thankyou. Updated